vue/sbc-common-components/src/services/fee.services.ts

```diff
@@ -1,58 +1,57 @@
 import Axios from 'axios'
-import { Fee } from '../models/fee'
+import { Fee, FilingData, PayData } from '../models'
 import ConfigHelper from '@/util/config-helper'
 import { SessionStorageKeys } from '@/util/constants'
 
-// const API_URL = 'https://mock-lear-tools.pathfinder.gov.bc.ca/rest/SBC+Pay+API+Reference/1.0.1'
 // sample Microcks URLs =
 //   https://mock-lear-tools.pathfinder.gov.bc.ca/rest/SBC+Pay+API+Reference/1.0.1/api/v1/fees/CP/OTANN
 //   https://mock-lear-tools.pathfinder.gov.bc.ca/rest/SBC+Pay+API+Reference/1.0.1/api/v1/fees/CP/OTADD
 const API_URL = 'https://pay-api-dev.pathfinder.gov.bc.ca/api/v1/fees'
-export default {
 
-  getFee (filingData: { filingDescription: string, filingTypeCode: string, waiveFees: boolean, entityType: string }[], payApiUrl: string)
-  : Promise<Fee[]> {
+export default {
+  getFee (filingData: FilingData[], payApiUrl: string) : Promise<Fee[]> {
     const token = ConfigHelper.getFromSession(SessionStorageKeys.KeyCloakToken)
+
     if (filingData.length < 1) {
       Promise.resolve()
     }
-    let promises = []
+
+    const promises = []
     for (const filing of filingData) {
       if (!filing.filingTypeCode) {
         Promise.resolve()
       }
-      var url = `${payApiUrl}fees/${filing.entityType}/${filing.filingTypeCode}?waiveFees=${filing.waiveFees ? filing.waiveFees : false}`
-
+      const url = `${payApiUrl}fees/${filing.entityType}/${filing.filingTypeCode}?waiveFees=${!!filing.waiveFees}`
       promises.push(Axios.get(url, { headers: { Authorization: `Bearer ${token}` } }))
     }
 
     return Axios.all(promises)
       .then(Axios.spread((...args) => {
         // customise the response here
         return args
-          .map(response => response.data)
+          .map(response => response.data as PayData)
           .map(data => {
-            let clientPassedDesc = filingData.find(fee => fee.filingTypeCode === data.filingTypeCode).filingDescription
-            // default the title if client has'nt passed this on
-            data.filingType = !clientPassedDesc ? data.filingType : clientPassedDesc
+            const thisItem = filingData.find(fd => fd.filingTypeCode === data.filingTypeCode)
+            // default the title if client hasn't passed this on
+            const filingType = (thisItem && thisItem.filingDescription) ? thisItem.filingDescription : data.filingType
```

Comment:
can use ts optional chaining here

Reply:
OK, I updated this in the latest commit.